### PR TITLE
[vscode] Improved Tree View

### PR DESCRIPTION
**Pull request by svenefftinge**
#### What it does
Fixes various smaller issues with tree views.

- handle undefined treeItem.label
- better styling for item actions
- support descriptions
- fix color for inactive selections (layout (grey) not accent (blue))
- don't execute command on selection change

#### How to test
Install gitlens extensions and compare the view with VS Code.

#### Review checklist

- [x] as an author, I have thoroughly tested my changes and carefully followed [the review guidelines](https://github.com/theia-ide/theia/blob/master/doc/pull-requests.md#requesting-a-review)

#### Reminder for reviewers

- as a reviewer, I agree to behave in accordance with [the review guidelines](https://github.com/theia-ide/theia/blob/master/doc/pull-requests.md#reviewing)

### CODE REVIEW

**Comment by akosyakov** [comment]
Does it address https://github.com/eclipse-theia/theia/issues/6142? 

It would be good to test with other tree views, just to check that we handle icons of different formats and sizes good. 

Plugin tree bugs were tested before against:
- https://github.com/microsoft/vscode-extension-samples/tree/master/tree-view-sample
- https://github.com/Azure/vscode-kubernetes-tools

**Comment by svenefftinge** [comment]
> Does it address #6142?

yes, I think so.

**Comment by svenefftinge** [comment]
I will dive in deeper tomorrow

**Comment by AlexTugarev** [comment]
'git it! the preference schema commit is necessary in order to test this PR, otherwise the gitlense extension is not loaded.

**Comment by AlexTugarev** [comment]
That's a big improvement!

Though the logs still look slightly reddish for the gitlense extension, the rendering of trees is much nicer!

File: packages/plugin/src/theia.d.ts [hunk 1]
```diff
@@ -3877,6 +3877,12 @@ declare module '@theia/plugin' {
          */
         iconPath?: string | Uri | { light: string | Uri; dark: string | Uri } | ThemeIcon;
 
+        /**
+         * A human readable string which is rendered less prominent.
+         * When `true`, it is derived from [resourceUri](#TreeItem.resourceUri) and when `falsy`, it is not shown.
+         */
+        description?: string | boolean;
```
Comment (AlexTugarev) [inline packages/plugin/src/theia.d.ts]: q: do we need a change log entry for this?

Comment (svenefftinge) [reply packages/plugin/src/theia.d.ts]: I don't think so. It is an addition of an optional property that actually should have been there before.

**Comment by svenefftinge** [comment]
To be clear: This is not fixing all issues the gitlens extension has with Theia. That would be a very big PR. Instead I focussed on the tree view stuff solely. I'm going to look into other issues in separate PRs.

**Comment by AlexTugarev** [comment]
Got it, thanks! These changes look good to me. Once you've removed the commit from pref schema PR, this is fine.

**Comment by svenefftinge** [comment]
I will fix the other PR so it can be merged and then rebase this on master

**Comment by AlexTugarev** [comment]
just verified,  #6142 is solved in this PR

**Pull request merged**